ddtrace/span.py

```diff
@@ -129,6 +128,11 @@ def __init__(
         if not (parent_id is None or isinstance(parent_id, six.integer_types)):
             raise TypeError("parent_id must be an integer")
 
+        self._tracer = None
+        if tracer is not None:
+            # accessing span.tracer logs a deprecation warning
+            self.tracer = tracer  # type: ignore[assignment]
```

Comment:
I had this before but `Tracer.trace()` creates a Span and passes the tracer as an argument. To avoid adding a breaking change in the next release I decided to leave that logic as is (I will remove the tracer arg from `Tracer._start_span` in the 1.0-dev branch). 

Making this change here will log a warning every time tracer.trace() is called. Is this acceptable? I could work around this by adding a `log_warning` parameter to Span.__init__() and setting it to true in `Tracer._start_span`

Reply:
I will set Span._tracer in `tracer.trace` instead. That approach makes more sense